minikube/resource_cluster.go

```diff
@@ -271,13 +267,18 @@ func initialiseMinikubeClient(d *schema.ResourceData, m interface{}) (service.Cl
 		return nil, err
 	}
 
+	apiserverNames := []string{}
+	if d.Get("apiserver_names").(*schema.Set).Len() > 0 {
+		apiserverNames = append(apiserverNames, d.Get("apiserver_names").(*schema.Set).List()[0].(string))
+	}
+
```

Comment:
It appears that apiserver_names can be multiple values i.e.

```console
minikube start --apiserver-names "foo.minikube.com" --apiserver-names "bar.minikube.com"
```

So I think we should keep the property as it is schema_cluster.go and instead 

```suggestion
	apiserverNames := []string{}
	if d.Get("apiserver_names").(*schema.Set).Len() > 0 {
		apiserverNames = d.Get("apiserver_names")
	}
```

and then cast it using the `ReadSliceState` function `APIServerNames:  state_utils.ReadSliceState(apiserverNames)` 

In terms of preferring a  `Set` over a `List`. This is mostly so we can avoid causing an accidental `diff` where the minikube cli would otherwise no-op. I think it's largely OK to allow duplicates to be configured, as we ultimately rely on what the underlying minikube lib allows (which does in fact to allow duplicate apiserver names apparently 😕)